src/cmd/index.js

```diff
@@ -1,8 +1,12 @@
 /* @flow */
-import build from './build';
-import lint from './lint';
-import run from './run';
-import sign from './sign';
-import docs from './docs';
 
-export default {build, lint, run, sign, docs};
+const defaultCommands = {};
+
+for (const cmd of ['build', 'lint', 'run', 'sign', 'docs']) {
+  defaultCommands[cmd] = async function(params: Object, options: Object) {
+    const {default: runCommand} = await import(`./${cmd}`);
```

Comment:
To be fair I'm not really thrilled about the webpack generated bundle chunks due to the dynamic import, especially numbered.

if we have to generate more then one transpiled file, it could be reasonable to just transpile everything file by file by using something like `babel-cli -d lib src/`, that at least would be less confusing (but it also would require a number of other changes, and I would prefer to avoid that for now).

Another option, if we agree on splitting the webpack bundle into chunks, would be to use set the webpack `output.chunkFilename` config option (https://webpack.js.org/configuration/output/#output-chunkfilename) to something like 'web-ext.[name].js' and then annotate the import using webpackChunkName webpack inline comment (https://webpack.js.org/api/module-methods/#import-), which would be possible only if paired with the more explicit approach described above, something like:

```js
export default {
  async build(
    params: BuildCmdParams, options: BuildCmdOptions
  ): Promise<ExtensionBuildResult> {
    const {default: cmd} = await import(
      /* webpackChunkName: "cmd-build" */
      './build');
    return cmd(params, options);
  },
```

One last option could be the "webpackMode" set to "eager", which would prevent webpack from generating any chunk:

```
for (const cmd of ['build', 'lint', 'run', 'sign', 'docs']) {
  defaultCommands[cmd] = async function(
    params: Object, options: Object
  ): Promise<any> {
    // $FLOW_FIXME: flow complains about any non literal module import or require.
    const {default: runCommand} = await import(
      /* webpackMode: "eager" */
      `./${cmd}`);
    return runCommand(params, options);
  };
}
```

which would be fine if we are ok with leave the type annotations of the functions exported by this module that open (I mean that `Object` is kind of "catch all" for the flow type checking, similarly to `Function` and `any`).

To be fair it doesn't matter when this module is used as an exported API to be used in other projects (because all the flow type are stripped when the sources are transpiled), and the type checking of the "program.js" module file is already pretty "relaxed" (but lets also take a look at the flow coverage more explicitly to evaluate the impact of it, one way to do it is using [flow-coverage-report](https://github.com/rpl/flow-coverage-report/) ;-))